src/Microsoft.OData.Core/InstanceAnnotationWriteTracker.cs

```diff
@@ -18,14 +18,13 @@ internal sealed class InstanceAnnotationWriteTracker
         /// Maintains the write status for each annotation using its key.
         /// If a key exists in the list then it is considered written.
         /// </summary>
-        private readonly HashSet<string> writeStatus;
+        private HashSet<string> writeStatus;
 
         /// <summary>
         /// Creates a new <see cref="InstanceAnnotationWriteTracker"/> to hold write status for instance annotations.
         /// </summary>
         public InstanceAnnotationWriteTracker()
         {
-            this.writeStatus = new HashSet<string>(StringComparer.Ordinal);

```

Comment:
Curious: could this be solved by a `Lazy` initializer?

Reply:
Yes it could be. The reason I opted for this approach is so that I only initialize the set when `MarkAsWritten` is called. If used the `Lazy` initializer, it would also be initialized when `IsAnnotationWritten` is called. I thought we could eliminate the initialization when `IsAnnotationWritten` is called since we can return false if the set is still null without having to initialize it.